### PR TITLE
Update CI to use Postgres 14

### PR DESCRIPTION
**Pull request by yajo**
@Tecnativa TT22772

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -32,14 +32,15 @@ jobs:
       fail-fast: false
       matrix:
         # Test modern Odoo versions with latest Postgres version
-        pg_version: ["12"]
+        pg_version: ["13"]
         odoo_version:
```
Comment (joao-p-marques) [inline .github/workflows/ci.yaml:36]: Shouldn't you add version 14.0 here? Or does that go on a different PR?

Comment (Yajo) [reply .github/workflows/ci.yaml:36]: Yes, that's in #327
